### PR TITLE
images from image host. +refactor

### PR DESCRIPTION
**Pull request by lbiedinger**
re EC-2357

Introduces a PictureVersionHelper in order to serve images from the image host directly instead of going through the Picture controller.
Also prefers symbols of string keys for the versions hash.
Better rubocop compliance

### CODE REVIEW

File: lib/europeana/mixins/image_version.rb [hunk 1]
```diff
@@ -30,16 +32,14 @@ module ImageVersion
 
       def versions(name = 'image')
         @versions ||= {}
-        @versions[name] ||= begin
+        @versions[name.to_sym] ||= begin
           image = @element.content_by_name(name)
           if image&.essence&.picture.present?
             versions_hash = {}
```
Comment (rwd) [inline lib/europeana/mixins/image_version.rb]: Let's replace:
```ruby
            versions_hash = {}
            VERSIONS.each_pair do |version, settings|
              picture = image.essence.picture
              url = version_url(picture_version(picture, settings))
              versions_hash[version] = { url: url }
            end
            versions_hash
```
with:
```ruby
            VERSIONS.each_with_object({}) do |(version, settings), memo|
              picture = image.essence.picture
              url = version_url(picture_version(picture, settings))
              memo[version] = { url: url }
            end
```

File: app/helpers/picture_version_helper.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    dragonfly_job = dragonfly_job.encode(settings[:format], options.compact.join(' '))
+    Alchemy::PictureVersion.from_cache(dragonfly_job)
+  end
+
+  ##
+  # Looks up the remote url for a Alchemy::PictureVersion, only works for S3DataStore.
+  #
+  # @param version_picture [Alchemy::PictureVersion]
+  def version_url(version)
```
Comment (rwd) [inline app/helpers/picture_version_helper.rb]: Could we rename this `picture_version_url` for clarity?

File: app/helpers/picture_version_helper.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    if Dragonfly.app(:alchemy_pictures).datastore.is_a?(Dragonfly::S3DataStore)
```
Comment (rwd) [inline app/helpers/picture_version_helper.rb]: Is it intentional (and useful) that this returns `nil` if not using S3?

File: lib/europeana/page.rb [hunk 4]
```diff
@@ -66,12 +68,12 @@ def media
...
 
     def as_chapter
-      {
+      @as_chapter ||= {
         is_chapter_nav: true,
         title: @page.title,
         url: show_page_url(@page.language_code, @page.urlname),
-        label: find_thumbnail ? find_thumbnail[:label] : false,
-        image: find_thumbnail ? find_thumbnail[:image] : false
+        label: chapter_thumbnail ? chapter_thumbnail[:label] : false,
```
Comment (rwd) [inline lib/europeana/page.rb]: `#chapter_thumbnail` always returns a hash. Either it should return `nil` when there is none, or the condition here (and on the following line) needs to be `chapter_thumbnail.present?`

File: lib/europeana/page.rb [hunk 5]
```diff
@@ -287,13 +289,18 @@ def exhibitions
...
-    def find_thumbnail
-      intro_element = page_elements.detect { |element| element.name == 'intro' }
-      return Europeana::Elements::ChapterThumbnail.new(intro_element).to_hash if intro_element
-
-      img_element = page_elements.detect { |element| %w(image rich_image credit_intro).include?(element.name) }
-      return Europeana::Elements::ChapterThumbnail.new(img_element).to_hash if img_element
-      false
+    def chapter_thumbnail
+      @chapter_thumbnail ||= begin
+        intro_element = page_elements.detect { |element| element.name == 'intro' }
```
Comment (rwd) [inline lib/europeana/page.rb]: Let's replace:
```ruby
        intro_element = page_elements.detect { |element| element.name == 'intro' }
        img_element = page_elements.detect { |element| %w(image rich_image credit_intro).include?(element.name) }
        if intro_element
          Europeana::Elements::ChapterThumbnail.new(intro_element).to_hash
        elsif img_element
          Europeana::Elements::ChapterThumbnail.new(img_element).to_hash
        else
          {}
        end
```
with:
```ruby
        if intro_element = page_elements.detect { |element| element.name == 'intro' }
          Europeana::Elements::ChapterThumbnail.new(intro_element).to_hash
        elsif img_element = page_elements.detect { |element| %w(image rich_image credit_intro).include?(element.name) }
          Europeana::Elements::ChapterThumbnail.new(img_element).to_hash
        else
          {}
        end
```
to save constructing `img_element` when not needed.

File: lib/europeana/page.rb [hunk 6]
```diff
@@ -355,26 +362,27 @@ def description
     end
 
     def thumbnail(version = :full)
-      if find_thumbnail && find_thumbnail.has_key?(:image) && find_thumbnail[:image]
-        return full_url(find_thumbnail[:image][version][:url])
+      if chapter_thumbnail&.key(:image)
```
Comment (rwd) [inline lib/europeana/page.rb]: Should this be `if chapter_thumbnail&.key?(:image)`?

File: lib/europeana/page.rb [hunk 7]
```diff
@@ -355,26 +362,27 @@ def description
...
+        return full_url(chapter_thumbnail[:image][version][:url])
       end
       false
     end
 
     def full_url(path)
-      "http://#{ENV.fetch('CDN_HOST', ENV.fetch('APP_HOST', 'localhost'))}#{ENV.fetch('APP_PORT', nil).nil? ? '' : ':'+ ENV.fetch('APP_PORT', nil)}#{path}"
+      port = ENV.fetch('APP_PORT', nil)
+      "http://#{ENV.fetch('CDN_HOST', ENV.fetch('APP_HOST', 'localhost'))}#{port.nil? ? '' : ':' + port}#{path}"
```
Comment (rwd) [inline lib/europeana/page.rb]: Could we use https:// (outside of localhost)?

File: app/helpers/picture_version_helper.rb (hunk 3 above)
Comment (lbiedinger) [reply app/helpers/picture_version_helper.rb]: the only real alternative currently in use would be a file store which doesn't provide URLs and would have to go through the pictures controller still.
So returning nil isn't ideal, but I don't really consider it worth it to support the pictures controller just for local deployments. I don't want to remove it just yet with this PR, but I think that is something simple enough that can happen as a follow up.

**Pull request merged**